callstack/react-native-builder-bob
# feat(legacy view template): extract view to a separate file

### Summary

Closes #370

Inspecting the codebase I've spotted that all of the view templates have the View initialized on a different file except the `legacy view` template. Having the view initialized in the `index.tsx` reduces the DX as anytime you make changes to the 'index.tsx` file, you will get `Invariant Violation: Tried to register two views with the same name` error. This PR addresses these issues.

### Test plan

1. Run `create-react-native-library` and select these options:
```
➜ ./create-react-native-library test
✔ What is the name of the npm package? … react-native-test
✔ What is the description for the package? … test
✔ What is the name of package author? … Burak Güner
✔ What is the email address for the package author? … md.burak.guner@gmail.com
✔ What is the URL for the package author? … https://github.com/atlj
✔ What is the URL for the repository? … https://github.com/atlj/react-native-test
✔ What type of library do you want to develop? › Native view
✔ Which languages do you want to use? › Kotlin & Swift
```
2. Go inside `src/`
3. Make sure `<Your Project Name>ViewNativeComponent.ts` exists with proper code
4. Make sure the view is imported and exposed in `index.tsx`
5. Make sure the app is able to display the view
6. Make sure when you make changes to `index.tsx`, the `Invariant Violation: Tried to register two views with the same name` error doesn't appear.
